packages/core/src/types.ts

```diff
@@ -20,6 +22,9 @@ export interface WalletConfig {
   id: string
   key: string
   keyDerivationMethod?: KeyDerivationMethod
+  storageType?: WalletStorageType
+  storageConfig?: WalletStorageConfig
+  storageCreds?: WalletStorageCreds
```

Comment:
Maybe we can chamge the interface of this to be not 100% what indy expects, but more how we would want the user of the framework to provide the values. Mainly with the addition of other wallet types I'm not sure if this approach fits nicely. I was thinking about something like this:

```ts
// defined in types.ts
export interface WalletConfig {
  id: string
  key: string
  keyDerivationMethod?: KeyDerivationMethod
  storage?: {
    type: string
    [key: string]: unknown
  }
}

// defined in @aries-framework/node package
interface IndyPostgresStorageConfig {
  type: 'postgres',
  config: {
    // .. parameters needed for postgres plugin
  },
   credentials: {
    // .. parameters needed for postgres plugin
  }
}
```

It can then be used like this:

```ts
import { IndyPostgresStorageConfig } from '@aries-framework/core'

const storageConfig: IndyPostgresStorageConfig = {
  type: 'postgres',
  // other required fields
}

await loadPostgresPlugin()

const agent = new Agent({
  walletConfig: {
     // other required fields
    storage: storageConfig
  }
})

await agent.initialize()
```